bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -875,37 +765,34 @@ func (gcp *Gcp) Generate(resources kftypes.ResourceEnum, options map[string]inte
 			return fmt.Errorf("could not generate files under %v Error: %v", K8S_SPECS, generateK8sSpecsErr)
 		}
 	case kftypes.PLATFORM:
-		gcpConfigFilesErr := gcp.generateDMConfigs(options)
+		gcpConfigFilesErr := gcp.generateDMConfigs()
 		if gcpConfigFilesErr != nil {
 			return fmt.Errorf("could not generate deployment manager configs under %v Error: %v", GCP_CONFIG, gcpConfigFilesErr)
 		}
 	}
-	email := options[string(kftypes.EMAIL)].(string)
-	nv := gcp.GcpApp.Spec.ComponentParams["cert-manager"]
-	setNameVal(&nv, "acmeEmail", email, true)
-	ipName := options[string(kftypes.IPNAME)].(string)
-	if ipName == "" {
-		ipName = gcp.GcpApp.Name + "-ip"
-	}
-	hostname := options[string(kftypes.HOSTNAME)].(string)
-	if hostname == "" {
-		hostname = gcp.GcpApp.Name + ".endpoints." + gcp.GcpApp.Spec.Project + ".cloud.goog"
-	}
-	if gcp.GcpApp.Spec.UseBasicAuth {
-		nv = gcp.GcpApp.Spec.ComponentParams["basic-auth-ingress"]
-		setNameVal(&nv, "ipName", ipName, true)
-		setNameVal(&nv, "hostname", hostname, true)
+	nv := gcp.Spec.ComponentParams["cert-manager"]
+	setNameVal(&nv, "acmeEmail", gcp.Spec.Email, true)
+	if gcp.Spec.IpName == "" {
+		gcp.Spec.IpName = gcp.Name + "-ip"
+	}
+	if gcp.Spec.Hostname == "" {
+		gcp.Spec.Hostname = gcp.Name + ".endpoints." + gcp.Spec.Project + ".cloud.goog"
+	}
+	if gcp.Spec.UseBasicAuth {
+		nv = gcp.Spec.ComponentParams["basic-auth-ingress"]
+		setNameVal(&nv, "ipName", gcp.Spec.IpName, true)
+		setNameVal(&nv, "hostname", gcp.Spec.Hostname, true)
 	} else {
-		nv = gcp.GcpApp.Spec.ComponentParams["iap-ingress"]
-		setNameVal(&nv, "ipName", ipName, true)
-		setNameVal(&nv, "hostname", hostname, true)
-	}
-	nv = gcp.GcpApp.Spec.ComponentParams["pipeline"]
-	setNameVal(&nv, "mysqlPd", gcp.GcpApp.Name+"-storage-metadata-store", false)
-	setNameVal(&nv, "minioPd", gcp.GcpApp.Name+"-storage-artifact-store", false)
-	createConfigErr := gcp.writeConfigFile(options)
+		nv = gcp.Spec.ComponentParams["iap-ingress"]
+		setNameVal(&nv, "ipName", gcp.Spec.IpName, true)
+		setNameVal(&nv, "hostname", gcp.Spec.Hostname, true)
+	}
+	nv = gcp.Spec.ComponentParams["pipeline"]
+	setNameVal(&nv, "mysqlPd", gcp.Name+"-storage-metadata-store", false)
+	setNameVal(&nv, "minioPd", gcp.Name+"-storage-artifact-store", false)
+	createConfigErr := gcp.writeConfigFile()
```

Comment:
Will we lose part of "app.yaml" data written during ksonnet.Init()?
As gcp.go read/write less fields compared with ksonnet.go